### PR TITLE
Use XDG-compliant paths for cache and config files

### PR DESCRIPTION
**Pull request by kephale**
Closes https://github.com/scenerygraphics/sciview/issues/452

### CODE REVIEW

File: src/main/kotlin/sc/iview/SciView.kt [hunk 1]
```diff
@@ -1796,6 +1797,17 @@ fun deleteNode(node: Node?, activePublish: Boolean = true) {
...
     }
 
+    /**
+     * Return ProjectDirectories for sciview.
+     *
+     * Use this to find the location of cache, etc.
+     *
+     * @return a [ProjectDirectories] for sciview
+     */
+    fun getProjectDirectories(): ProjectDirectories {
```
Comment (ctrueden) [inline src/main/kotlin/sc/iview/SciView.kt:1807]: We might want to gather all sciview options into an options/config class, accessible via `sciview.getOptions()` or some such. This class could have the `ProjectDirectories` and also have any other state for managing configuration. What do you think? Is there significant other configuration to be managed?

**Review by ctrueden — APPROVED** [review]
:+1: :+1: in support of this. I just was wondering whether we should make a new class for the sciview options, to help minimize the size of `SciView.kt`, and group conceptually related things together. But I'm happy either way; this PR is still a step forward.

File: src/main/kotlin/sc/iview/SciView.kt (hunk 1 above)
Comment (kephale) [reply src/main/kotlin/sc/iview/SciView.kt:1807]: Mouse and key config is blurred between scenery and sciview, which is part of the issue with folks current input config concerns. I'm looking at the most surgical way of ensuring consistent input controls, but there are a lot of layers here...

**Comment by kephale** [comment]
Ok, I'm going to merge for now. I created a tracking issue for the idea of creating an overall config/options class: https://github.com/scenerygraphics/sciview/issues/507

**Pull request merged**